pkgs/by-name/li/libphidget22extra/package.nix

```diff
@@ -7,15 +7,15 @@
   libphidget22,
 }:
 let
-  version = "1.20.20240909";
+  version = libphidget22.version;
```

Comment:
Is this a good approach for making sure that they share the same version?

Reply:
Not exactly, since we would be more likely to forget updating the hash.

Reply:
> Not exactly, since we would be more likely to forget updating the hash.

Shouldn't it recalculate the hash when the url is changed? 
I've noticed that in some cases it ignores the hash and uses the old one. when updating a package, I have to change the hash to an empty string to get the new one. why does it behave this way?

Reply:
In short: these `src` derivations are a special type of derivation called "fixed output derivation" (or FOD), they are only ever get rebuilt if their output hashes change (that is the `hash` part).

Reply:
makes sense now :+1: 

What do you suggest? should we manually handle the version string like before?

Reply:
> should we manually handle the version string like before?

Yes, but leave a comment on the side of the version number, reminding everyone to update these two together, as in `pkgs/development/libraries/grpc/default.nix`